update_image.pl

```diff
@@ -359,6 +370,34 @@ sub processConvergedSections {
     }
     $sections{HDAT}{out}       = "$scratch_dir/hdat.bin.ecc";
 
+    # Populate OCMBFW partition if it exists in the layout
+    if(checkForPnorPartition("OCMBFW", $parsed_pnor_layout))
+    {
+        if(-e $ocmbfw_original_filename)
+        {
+            #Verify image validity
+            run_command("$hb_image_dir/pkgOcmbFw.pl --verify --packagedBin $ocmbfw_original_filename");
+            $sections{OCMBFW}{in}    = "$ocmbfw_original_filename";
+        }
+        else
+        {
+            print "WARNING: OCMBFW binary not found, generating blank binary (w/ valid header) instead\n";
+            #Create blanke 4k image
+            my $ocmbfw_generated_filename = "ocmbfw_generated.bin";
+            run_command("dd if=/dev/zero of=$scratch_dir/$ocmbfw_generated_filename bs=1024 count=4");
+        
+            #Add header to blank image
+            my $date = `date`;
+            run_command("$hb_image_dir/pkgOcmbFw.pl --unpackagedBin $scratch_dir/$ocmbfw_generated_filename --packagedBin $ocmbfw_original_filename --timestamp \"$date\" --vendorVersion \"0.1\" --vendorUrl \"http://www.ibm.com\"");
```

Comment:
this line uses $ocmbfw_original_filename, but I don't think this has been set yet.

Same issue for lines 394 and 395 below

Reply:
Right -- that is in the other PR I have you on review for here: https://github.com/open-power/op-build/pull/2929

Reply:
but your else branch doesn't work unless you set something to $ocmbfw_original_filename

Reply:
Right, ocmbfw_original_filename will either be the default filename -- or if somebody set a different name in the defconfig file -- it'll point to that one. The check above says -- does this file exist -- meaning is anybody populating it. If not we'll generate some 'valid' empty one for now. If it does exist (this basically implies some other package has built an ocmbfw image), then lets use that.

Reply:
Yep.  I totally misread your code and the use of the "-e" in the check.  My bad.